### PR TITLE
Tweaks to MetricsDescriptorImpl

### PR DESCRIPTION
**Pull request by viliam-durina**
- reduce complexity in buildMetricString(): It's less branching to replace the
comma at the end than to check if we need to add a comma before each item.

- throw when index out of bounds instead of returning null

- more tests

- multiple trivial changes

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/internal/metrics/impl/MetricDescriptorImplTest.java [hunk 1]
```diff
@@ -68,6 +69,27 @@ public void testConstructorWithNullThrows() {
         new MetricDescriptorImpl(null);
     }
 
+    @Test
+    public void testTabByIndex() {
+
```
Comment (blazember) [inline hazelcast/src/test/java/com/hazelcast/internal/metrics/impl/MetricDescriptorImplTest.java]: Empty test case

**Comment by viliam-durina** [comment]
verify

**Comment by viliam-durina** [comment]
verify

**Comment by viliam-durina** [comment]
verify

**Comment by viliam-durina** [comment]
run-lab-run

**Comment by blazember** [comment]
Trivial changes, merging with 1 review.

**Pull request merged**